stdlib/Test/src/Test.jl

```diff
@@ -312,7 +312,9 @@ function eval_test(evaluated::Expr, quoted::Expr, source::LineNumberNode, negate
 
     Returned(res,
              # stringify arguments in case of failure, for easy remote printing
-             res === true ? quoted : sprint(io->print(IOContext(io, :limit => true), quoted))*kw_suffix,
+             res === true ? quoted : let quoted=quoted
+                     sprint(io::IOBuffer->print(IOContext(io, :limit => true), quoted))*kw_suffix
+		 end,
```

Comment:
```suggestion
             res === true ? quoted : sprint(print, quoted, context=(:limit => true)) * kw_suffix),
```

Also, in general, it can be simplified by use of the varargs ability, though not needed here:
```suggestion
             res === true ? quoted : sprint((io::IOBuffer, q) -> print(IOContext(io, :limit => true), q), quoted) * kw_suffix,
```
(though I think this also this line should have been pulled out into a variable, so that the nesting on this expression wasn't quite so extreme)